src/_P001_Switch.ino

```diff
@@ -173,51 +244,152 @@ boolean Plugin_001(byte function, struct EventStruct *event, String& string)
+            {
+              case PLUGIN_001_BUTTON_TYPE_NORMAL_SWITCH:
+                  new_outputState = state;
+                break;
+              case PLUGIN_001_BUTTON_TYPE_PUSH_ACTIVE_LOW:
+                if (!state)
+                  new_outputState = !currentOutputState;
+                break;
+              case PLUGIN_001_BUTTON_TYPE_PUSH_ACTIVE_HIGH:
+                if (state)
+                  new_outputState = !currentOutputState;
+                break;
+            }
+
+            // send if output needs to be changed
+            if (currentOutputState != new_outputState)
+            {
+              byte output_value;
+              outputstate[event->TaskIndex] = new_outputState;
+              boolean sendState = new_outputState;
+              if (Settings.TaskDevicePin1Inversed[event->TaskIndex])
+                sendState = !sendState;
+
+              if (Plugin_001_clickCounter[event->TaskIndex]==3 && Settings.TaskDevicePluginConfig[event->TaskIndex][5])
+              {
+                output_value = 3; //double click
+              } else {
+                output_value = sendState ? 1 : 0; //single click
               }
+              event->sensorType = SENSOR_TYPE_SWITCH;
+              if (P001_getSwitchType(event) == PLUGIN_001_TYPE_DIMMER) {
+                if (sendState) {
+                  output_value = Settings.TaskDevicePluginConfig[event->TaskIndex][1];
+                  // Only set type to being dimmer when setting a value else it is "switched off".
+                  event->sensorType = SENSOR_TYPE_DIMMER;
+                }
+              }
+              UserVar[event->BaseVarIndex] = output_value;
+              String log = F("SW   : Switch state ");
+              log += state ? F("1") : F("0");
+              log += F(" Output value ");
+              log += output_value;
+              addLog(LOG_LEVEL_INFO, log);
+              sendData(event);
+            }
+            Plugin_001_clickTimePrevious[event->TaskIndex] = millis();
+          }
+        }
+        //check if LP is enabled and if LP has not fired yet
+        else if (Settings.TaskDevicePluginConfig[event->TaskIndex][7] && !Plugin_001_firedLP[event->TaskIndex]) {
+          /**************************************************************************\
+          20181009 - @giig1967g: new longpress logic is:
+          if there is no 'state' change, check if longpress interval reached
+          When reached send longpress event.
+          Returned Event value = state + 10
+          So if state = 0 => EVENT longpress = 10
+          if state = 1 => EVENT longpress = 11
+          So we can trigger longpress for high or low contact
+
+          In rules this can be checked:
+          on Button#Switch=10 do //will fire if longpress when state = 0
+          on Button#Switch=11 do //will fire if longpress when state = 1
+          \**************************************************************************/
+
```

Comment:
Because it's not necessary.
Now the switch returns 0 or 1 when there is a state change.
I added 3, 10 and 11 to indicate doubleclick, longpress high and longpress low.
See coding example in the first reply.